project.mml

```diff
@@ -475,12 +475,12 @@ Layer:
             ELSE NULL
           END AS maxspeed_kmh,
           CASE
-            WHEN (
-              tags->'psv' != 'no' OR tags->'motorcar' != 'no' OR tags->'bus' != 'no'
-              OR tags->'motor_vehicle' != 'no'
-              OR tags->'vehicle' != 'no'
-              ) THEN 'yes'
-            WHEN tags->'psv' IS NULL AND tags->'motorcar' IS NULL AND tags->'bus' IS NULL AND tags->'motor_vehicle' IS NULL AND tags->'vehicle' IS NULL AND (access != 'no' OR access IS NULL) THEN 'yes'
+            WHEN COALESCE(motorcar, tags->'motor_vehicle', tags->'vehicle', access, 'yes') != 'no' THEN 'yes'
+            -- goods and hgv are not technically children of motorcar, but the motorcar restriction is often intended to cover cars and all larger vehicles
+            WHEN COALESCE(tags->'goods', motorcar, tags->'motor_vehicle', tags->'vehicle', access, 'yes') != 'no' THEN 'yes'
```

Comment:
Couldn't this be reduced to a simple check on `tags->'goods'? The remaining should have already been handled by the first `WHEN` case.

Reply:
Yeah, it could, but I thought leaving the full expression would make it easier to understand the logic (and to copy-paste if someone comes in in the future).

Reply:
Please use the simplest version, we should taking care of time performance